internal/resources/helmrelease/spec/spec.go

```diff
@@ -237,3 +257,36 @@ func HasSpecChanged(d *schema.ResourceData) bool {
 
 	return updateRequired
 }
+
+func readYamlFile(fileName string) (string, error) {
+	inputFile, err := os.Open(fileName)
+	if err != nil {
+		return "", errors.WithMessage(err, fmt.Sprintf("Error opening the %s file.", fileName))
+	}
+
+	defer inputFile.Close()
+
+	buf := bytes.NewBuffer(nil)
+	_, err = io.Copy(buf, inputFile)
+
+	if err != nil {
+		return "", err
+	}
+
+	_, err = yaml.Marshal(buf.String())
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func fileExists(filepath string) bool {
+	fileinfo, err := os.Stat(filepath)
+
+	if os.IsNotExist(err) {
+		return false
+	}
+
+	return !fileinfo.IsDir()
+}
```

Comment:
Can you log the errors? Please log at other places too.

Reply:
Updated PTAL